#! /bin/sh -e
## octave-2.1.53-demos.dpatch by Rafael Laboissiere <rafael@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Use warn_empty_list_elements instead of empty_list_elements_ok

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

if [ $# -ne 1 ]; then
  echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
  exit 1
fi
case "$1" in
  -patch)
    patch $patch_opts -p1 < $0;;
  -unpatch)
    patch $patch_opts -p1 -R < $0;;
  *)
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1;;
esac

exit 0
--- plplot-5.3.0.cvs.20040430.orig/bindings/octave/PLplot/support/__pl_plotit.m
+++ plplot-5.3.0.cvs.20040430/bindings/octave/PLplot/support/__pl_plotit.m
@@ -25,8 +25,11 @@
     return
   endif
 
-  old_empty_list_elements_ok = empty_list_elements_ok;
-  empty_list_elements_ok = 1;
+  old_empty_list_elements_ok = warn_empty_list_elements;
+
+  unwind_protect
+    
+  warn_empty_list_elements = 0;
 
   if (__pl.type(strm) >= 100 && __pl.type(strm) < 200)
     __pl_meshplotit;
@@ -132,7 +135,6 @@
     
     if (is_matrix(x) && !is_vector(x) && is_matrix(y) && !is_vector(y)) 
       if (style != 9 && (xc != yc || rows(x) != rows(y)))
-	empty_list_elements_ok = old_empty_list_elements_ok;
 	error ("__pl_plotit: matrix dimensions must match.");
       endif
       range = "i;";	
@@ -230,7 +232,6 @@
 
 	  case (9) ## errorbars	    
 	    if ( (xc == 1 && yc == 1) || xc > 3 || yc > 3)
-	      empty_list_elements_ok = old_empty_list_elements_ok;
 	      error("plot with errorbars: either x or y or both must be 2 or 3 columns.\n\
 		  If x (or y) has two columns, then it is interpreted as (x, dx),\n\
 		  and a bar is plotted from x-dx to x+dx;\n\
@@ -278,6 +279,10 @@
   pllab(tdeblank(__pl.xlabel(strm,:)), tdeblank(__pl.ylabel(strm,:)), tdeblank(__pl.tlabel(strm,:)));
   plflush;
 
-  empty_list_elements_ok = old_empty_list_elements_ok;
+  unwind_protect_cleanup  
+  
+  warn_empty_list_elements = old_empty_list_elements_ok;
+
+  end_unwind_protect  
 
 endfunction
